pkg/image/qemu.go

```diff
@@ -43,19 +50,34 @@ type QEMUOperations interface {
 
 type qemuOperations struct{}
 
-var qemuExecFunction = system.ExecWithLimits
-
-var qemuLimits = &system.ProcessLimitValues{AddressSpaceLimit: maxMemory, CPUTimeLimit: maxCPUSecs}
+var (
+	qemuExecFunction = system.ExecWithLimits
+	qemuLimits       = &system.ProcessLimitValues{AddressSpaceLimit: maxMemory, CPUTimeLimit: maxCPUSecs}
+	qemuIterface     = NewQEMUOperations()
+	re               = regexp.MustCompile(matcherString)
+
+	progress = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "import_progress",
+			Help: "The import progress in percentage",
+		},
+		[]string{"ownerUID"},
+	)
+	ownerUID string
+)
 
-var qemuIterface = NewQEMUOperations()
+func init() {
+	prometheus.MustRegister(progress)
+	ownerUID, _ = util.ParseEnvVar(common.OwnerUID, false)
+}
 
 // NewQEMUOperations returns the default implementation of QEMUOperations
 func NewQEMUOperations() QEMUOperations {
 	return &qemuOperations{}
 }
 
 func (o *qemuOperations) ConvertQcow2ToRaw(src, dest string) error {
-	_, err := qemuExecFunction(qemuLimits, "qemu-img", "convert", "-p", "-f", "qcow2", "-O", "raw", src, dest)
+	_, err := qemuExecFunction(qemuLimits, reportProgress, "qemu-img", "convert", "-p", "-f", "qcow2", "-O", "raw", src, dest)
```

Comment:
Yeah, also have to consider when we are not doing QEMU conversions, like the tar extractions we are doing soon. Maybe we should introduce the concept of phases? In case there are more than one, and report the progress for each phase?